src/settings.c

```diff
@@ -171,14 +177,19 @@ int git_libgit2_opts(int key, ...)
 		{
 			const char *file = va_arg(ap, const char *);
 			const char *path = va_arg(ap, const char *);
-			if (!SSL_CTX_load_verify_locations(git__ssl_ctx, file, path)) {
-				giterr_set(GITERR_NET, "SSL error: %s",
-					ERR_error_string(ERR_get_error(), NULL));
-				error = -1;
-			}
+			error = git_openssl_set_cert_file(file, path);
+		}
+#elif GIT_MBEDTLS
+		{
+			const char *file = va_arg(ap, const char *);
+			const char *path = va_arg(ap, const char *);
+			if (file)
+				error = git_mbedtls_set_cert_file(file, 0);
+			if (error && path)
+				error = git_mbedtls_set_cert_file(path, 0);
```

Comment:
Just want to point out, this weird call is due to the [insanity](https://www.openssl.org/docs/man1.0.2/ssl/SSL_CTX_load_verify_locations.html) of OpenSSL. This will cause verification checks to use the first valid certificate from `file` it finds, else it will enumerate cert files under `path`.
I'm afraid mbedTLS has no mechanism to provide a default CA store because it is out-of-scope for them (same issue with 0c7dd1e).